src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -399,14 +399,14 @@ private function describeValue($value, $omitTags, $showArguments)
             );
         }
 
-        if ($value instanceof Definition) {
-            return $this->getContainerDefinitionData($value, $omitTags, $showArguments);
-        }
-
         if ($value instanceof ArgumentInterface) {
             return $this->describeValue($value->getValues(), $omitTags, $showArguments);
         }
 
+        if ($value instanceof Definition) {
```

Comment:
changing the order here is because of the needs of https://github.com/symfony/symfony/pull/21708, right ? IMO, this is a clear sign that your other PR breaks BC (other bundles may have similar code)